hydra-cluster/test/Test/EndToEndSpec.hs

```diff
@@ -370,6 +375,75 @@ spec = around showLogsOnFailure $ do
               waitForLog 10 nodeStdout "JSON object with key NodeOptions" $ \line ->
                 line ^? key "message" . key "tag" == Just (Aeson.String "NodeOptions")
 
+      it "detects misconfiguration" $ \tracer -> do
+        withTempDir "temp-dir-to-check-hydra-misconfiguration" $ \dir -> do
+          withCardanoNodeDevnet (contramap FromCardanoNode tracer) dir $ \node@RunningNode{nodeSocket} -> do
+            hydraScriptsTxId <- publishHydraScriptsAs node Faucet
+            let persistenceDir = dir </> "persistence"
+            let cardanoSK = dir </> "cardano.sk"
+            let hydraSK = dir </> "hydra.sk"
+
+            (_, cardanoSKey) <- generateCardanoKey
+            hydraSKey :: SigningKey HydraKey <- generate arbitrary
+
+            void $ writeFileTextEnvelope hydraSK Nothing hydraSKey
+            void $ writeFileTextEnvelope cardanoSK Nothing cardanoSKey
+
+            -- generate node state to save to a file
+            openState :: OpenState Tx <- generate arbitrary
+            let headParameters = parameters openState
+                stateContestationPeriod = contestationPeriod headParameters
+
+            -- generate one value for CP different from what we have in the state
+            differentContestationPeriod <- generate $ arbitrary `suchThat` (/= stateContestationPeriod)
+
+            -- alter the state to trigger the errors
+            let alteredNodeState =
+                  Open
+                    ( openState
+                        { parameters =
+                            headParameters{contestationPeriod = differentContestationPeriod, parties = []}
+                        }
+                    )
+
+            -- save altered node state
+            void $ do
+              createDirectoryIfMissing True persistenceDir
+              BSL.writeFile (persistenceDir </> "state") (Aeson.encode alteredNodeState)
+
+            let nodeArgs =
+                  Options.toArgs
+                    Options.defaultRunOptions
+                      { Options.chainConfig =
+                          Options.defaultChainConfig
+                            { Options.cardanoSigningKey = cardanoSK
+                            , Options.nodeSocket
+                            , Options.contestationPeriod = contestationPeriod headParameters
+                            }
+                      , Options.hydraSigningKey = hydraSK
+                      , Options.hydraScriptsTxId
+                      , Options.persistenceDir
+                      , Options.ledgerConfig =
+                          Options.defaultLedgerConfig
+                            { Options.cardanoLedgerGenesisFile = "config/devnet/genesis-shelley.json"
+                            , Options.cardanoLedgerProtocolParametersFile = "config/protocol-parameters.json"
+                            }
+                      }
```

Comment:
Better, but why all the qualified identifiers? In case you get conflicts, `DuplicateRecordFields` should help.